### PR TITLE
debt: extract a buncha stuff from cmd/frontend that was used outside of frontend

### PR DESCRIPTION
**Pull request by Strum355**
As part of https://github.com/sourcegraph/sourcegraph/pull/52521, I had to investigate the possible binary targets that would require explicit calls to any of the new `Init`s that were previously implicit `init`s. As part of this, by the power of bazel, I discovered some dependencies on the changed packages in binaries that shouldnt need to (migrator depending on `cmd/frontend/internal/highlight`[1]??). So heres a mini-crusade to extract _some_ (but not all, because theres A LOT) of the things out `cmd/frontend/internal` into more appropriate packages (like `internal/`

<details>
<summary>[1]</summary>

```
$ bazel query 'kind("go_binary", rdeps(//..., //cmd/frontend/internal/highlight))'
//cmd/frontend:frontend
//cmd/sourcegraph-oss:sourcegraph-oss
//enterprise/cmd/frontend:frontend
//enterprise/cmd/migrator:migrator
//enterprise/cmd/sourcegraph:sourcegraph
//enterprise/cmd/worker:worker
```

</details>

## Test plan

https://buildkite.com/sourcegraph/sourcegraph/builds/223014

:fire: :slightly_smiling_face:

### CODE REVIEW

File: internal/auth/userpasswd/config_watch.go [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
 import (
-	"github.com/sourcegraph/sourcegraph/cmd/frontend/auth/providers"
+	"github.com/sourcegraph/sourcegraph/internal/auth/providers"
 	"github.com/sourcegraph/sourcegraph/internal/conf"
 )
 
 // Watch for configuration changes related to the builtin auth provider.
-func init() {
+func Init() {
```
Comment (efritz) [inline internal/auth/userpasswd/config_watch.go:9]: Is this called now?

Comment (Strum355) [reply internal/auth/userpasswd/config_watch.go:9]: Its called in frontend and in worker (can't link coz mobile)

**Review by efritz — APPROVED** [review]

**Review by camdencheek — APPROVED** [review]
This is great. I've had this stuff on my "maybe eventually" backlog forever. Thank you! 💯

**Comment by Strum355** [comment]
> This is great. I've had this stuff on my "maybe eventually" backlog forever. Thank you! 💯

Theres soooo much more left 😅 great excuse to learn bazel query if you havent already!

**Pull request merged**